File: ios/Shared/RelaySelector.swift
import Foundation

public struct RelaySelector: Sendable {
    public init() {}

    public func orderedCandidates(from relays: [RelayDescriptor], now: Date = Date()) -> [RelayDescriptor] {
        relays.filter { $0.isUsable(now: now) }
    }
}

File: android/app/src/main/java/com/openrung/model/RelaySelector.kt
package com.openrung.model

import java.time.Instant

class RelaySelector {
    fun orderedCandidates(relays: List<RelayDescriptor>, now: Instant): List<RelayDescriptor> =
        relays.filter { it.isUsable(now) }

    fun selectFirstUsable(relays: List<RelayDescriptor>, now: Instant): RelayDescriptor? =
        orderedCandidates(relays, now).firstOrNull()
}
